### PR TITLE
YieldFrom Smell

### PR DESCRIPTION
**Pull request by AdamGold**
add YieldFrom smell - replace ```for x in iterable yield x``` in ```yield from iterable```
# Provide links to any relevant tickets, articles or other resources
Closes: #51
Progresses : #

### CODE REVIEW

File: good_smell/smells/yield_from.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    def visit_For(self, node: ast.For):
+        yield_from = ast.Expr(
+            value=ast.YieldFrom(node.iter)
+        )
+        return ast.fix_missing_locations(yield_from)
+
+    @staticmethod
+    def is_smelly(node: ast.AST):
+        """Check if the node is only a nested for"""
+        # TODO check python version, check if inside a function
```
Comment (Tadaboody) [inline good_smell/smells/yield_from.py]: Yielding outside a function results in a SyntaxError, fixing that is outside this projects' scope. So I think it's valid to assume that if you see a yield, it's inside a function

Comment (Tadaboody) [reply good_smell/smells/yield_from.py]: Please check python version by adding a function to LintSmell which checks if it should run at all - Adding a check in `is_smelly` will run the check for _every node_ which is redundant

File: good_smell/smells/yield_from.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    def warning_message(self):
+        return "Consider using yield from instead of yield inside of a for loop"
+
+    @property
+    def symbol(self):
+        return "yield-from"
+
+
+class YieldFromTransformer(LoggingTransformer):
+    """NodeTransformer that goes visits all the nested `for`s and replaces them
```
Comment (Tadaboody) [inline good_smell/smells/yield_from.py]: Update the docstring

**Comment by AdamGold** [comment]
- [ ] Check Python version
- [ ] Visit yields and not fors then check parent?
- [ ] Pass tests

**Comment by Tadaboody** [comment]
I don't think checking the parents would be possible because there isn't a pointer to them. Maybe subclassing the ast to add them would prove worth it.
Would make #6 easier

**Comment by Tadaboody** [comment]
Where did you rebase to? Dropwhile was implemented in #56 and wasn't merged anywhere

**Comment by AdamGold** [comment]
> Where did you rebase to? Dropwhile was implemented in #56 and wasn't merged anywhere

Fixed

**Comment by Tadaboody** [comment]
🎉 🎉 Thanks for the contribution 🎉 🎉
Ship it

**Pull request merged**